src/main/java/com/github/splendor_mobile_game/game/model/Deck.java

```diff
@@ -0,0 +1,49 @@
+package com.github.splendor_mobile_game.game.model;
+
+import java.util.ArrayList;
+
+import com.github.splendor_mobile_game.game.enums.CardTier;
+
+public class Deck extends ArrayList<Card> {
+    private final CardTier tier;
+    private final int maxSize;
+
+    public Deck(CardTier tier){
+        super();
+        this.tier=tier;
+        this.maxSize=whatSize(tier);
+    }
+    public Deck(CardTier tier,int size){
+        super(size);
+        this.tier=tier;
+        this.maxSize=whatSize(tier);
+    }
+
+    public Deck(CardTier tier,java.util.Collection<? extends Card> array){
+        super(array);
+        this.tier=tier;
+        this.maxSize=whatSize(tier);
+    }
+
+    public CardTier GetTier(){
+        return tier;
+    }
+
+    public int MaxSize(){
+        return maxSize;
+    }
+
+    public static int whatSize(CardTier tier){
```

Comment:
maxSize is determined by database size. Instruction says what exact amount of different tiers cards is. getting or setting size of Deck is useless since `InMemoryDatabase` is holding right amount of all cards. It's duplicate of code and problem maker